app/src/functions/send-discord-notification/app.ts

```diff
@@ -0,0 +1,106 @@
+import { getQueryResults } from '../../lib/datadog';
+import { ApplicationLog, isApplicationLog, LambdaLog, LoadBalancerLog } from '../../models/datadog';
+import { getHourFromUtcDate, getTodayStringTime } from '../../lib/date-utils';
+import { formatNumber } from '../../utils/helpers';
+import { sendEmbedMessage, sendMessage } from '../../lib/discord';
+import { EmbedFieldData } from 'discord.js';
+
+type availableLogs = LoadBalancerLog | ApplicationLog
+
+// Goes through all the values and constantly updates the map with the most recent one,
+// as the logs already come sorted, thereby only keeping the latest log of each hour
+function filterMinimunDuplicates<T extends LambdaLog>(logs: T[]): T[] {
+  const filtered: T[] = []
+
+  const filter = new Map<string, T>()
+  logs.forEach(lb => filter.set(lb.hourstamp, lb))
+
+  for (const [_, lb] of filter.entries()) {
+    filtered.push(lb)
+  }
+
+  return filtered
+}
+
+// Returns a map of exceeded logs based on their hourstamp 
+function mapsExceededThresholds(logs: availableLogs[]): Map<string, availableLogs[]> {
+  const logsMap = new Map<string, availableLogs[]>()
+
+  logs.forEach(log => logsMap.set(log.id,
+    [...(logsMap.get(log.id) || []), log]))
+
+  for (const [id, logs] of logsMap.entries()) {
+    const filtered = filterMinimunDuplicates(logs)
+    logsMap.set(id, filtered)
+  }
+  return logsMap
+}
+
+function buildEmbedMessages(data: Map<string, availableLogs[]>): Map<string, EmbedFieldData[]> {
+  const messages = new Map<string, EmbedFieldData[]>()
+
+  for (const [_, logs] of data) {
+    const message: EmbedFieldData[] = []
+    if (isApplicationLog(logs[0])) {
+      const { applicationName: name, applicationPublicKey: publicKey, applicationAddress: adddress, email } = logs[0]
+      message.push(
+        { name: 'Public Key', value: publicKey, inline: false },
+        { name: 'Address', value: adddress, inline: true },
+        { name: "Email", value: email, inline: true })
+      for (const log of logs) {
```

Comment:
Could we add a space between these two lines? It feels like they're supposed to be the same logic when they aren't.